Fix hdmaster-key / seed-key confusion (scripted diff)

jnewbery opened the pull request:
Addresses #12084 and #8684

This renames a couple of functions and members (no functional changes, expect log prints):

- Rename CKey::SetMaster to CKey::SetSeed
- Rename CHDChain::masterKeyId to CHDChain::seedID
- Rename CHDChain::hdMasterKeyID to CHDChain::hdSeedID
- Rename CWallet::GenerateNewHDMasterKey to CWallet::GenerateNewHDSeed
- Rename CWallet::SetHDMasterKey to CWallet::SetHDSeed

As well it introduces a tiny API change:

- RPC API change: Rename "hdmasterkeyid" to "hdseedid", rename "hdmaster" in wallet-dump output to "hdseed"

Fixes also a bug:

- Bugfix: use "s" instead of the incorrect "m" for the seed-key hd-keypath key metadata

jnewbery committed:
scripted-diff: Rename master key to seed

-BEGIN VERIFY SCRIPT-

ren() { git grep -l "\<$1\>" 'src/*.cpp' 'src/*.h' test | xargs sed -i "s:\<$1\>:$2:g"; }
ren GenerateNewHDMasterKey  GenerateNewSeed
ren DeriveNewMasterHDKey    DeriveNewSeed
ren SetHDMasterKey          SetHDSeed
ren hdMasterKeyID           hd_seed_id
ren masterKeyID             seed_id
ren SetMaster               SetSeed
ren hdmasterkeyid           hdseedid
ren hdmaster                hdseed

-END VERIFY SCRIPT-

jnewbery committed:
[refactor] manually change remaining instances of master key to seed.

jnewbery commented:
This is #12094 reimplemented as a scripted diff.

practicalswift commented:
Concept ACK

laanwj commented:
Concept ACK

jonasschnelli commented:
Needs rebase

jnewbery committed:
[docs] Add release notes for HD master key -> HD seed rename

jnewbery commented:
Rebased and added release notes.

MarcoFalke commented:
Would it make sense to additionally push back the "hdmasterkeyid" key-value pair for one release? This might make it easier to quickly downgrade after a failed upgrade to the latest release.

jnewbery commented:
@MarcoFalke - sorry I don't fully understand. Can you point out in the code where you mean?

MarcoFalke commented on src/wallet/rpcwallet.cpp:
```diff
@@ -3841,7 +3841,7 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
         ret.pushKV("timestamp", meta->nCreateTime);
         if (!meta->hdKeypath.empty()) {
             ret.pushKV("hdkeypath", meta->hdKeypath);
-            ret.pushKV("hdmasterkeyid", meta->hdMasterKeyID.GetHex());
```
E.g. here it seems that you are removing `hdmasterkeyid` from the dict. To make it easier for people that rely on this key to upgrade/downgrade without having to change their code (or configuration) every time they switch, it could make sense to leave the `ret.pushKV("hdmasterkeyid", meta->hdMasterKeyID.GetHex());` in for now. (Can be removed in the next release, for example)

jnewbery replied:
Ah, I see. And also have `hdseedid` return the same value? Yes, I can do that.

I can also hide `hdmasterkeyid` behind a deprecation switch if you think that's useful.

MarcoFalke replied:
Not sure if we need the overkill of a deprecation switch for every tiny api change. Just returning both and mentioning that one of the is deprecated in the docs seems pragmatic.

jnewbery replied:
Sounds good. I'll just return both and mention in the release notes

The deprecation switch isn't too heavy, so I can add it if anyone thinks it's necessary I can add it.

jnewbery committed:
[rpc] [wallet] Add 'hdmasterkeyid' alias return values.

Restores the  return value in getwalletinfo() and getaddressinfo()
RPC methods for backwards compatibility

jnewbery commented:
@MarcoFalke - addressed your review comment.

MarcoFalke commented:
Needs rebase due to #12560?

MarcoFalke commented:
In the scripted diff you might want to remove the line `ren MasterKeyID             SeedID`, since it is unused. (Also, switching everything to camel case seems appropriate.)

utACK dead17b

jnewbery commented:
> In the scripted diff you might want to remove the line ren MasterKeyID SeedID, since it is unused. (Also, switching everything to camel case seems appropriate.)

Thanks. Both done!

MarcoFalke commented on test/functional/wallet_dump.py:
```diff
@@ -39,7 +39,7 @@ def read_dump(file_name, addrs, script_addrs, hd_master_addr_old):
                     if keytype == "inactivehdmaster=1":
```
This one should be fixed up in commit
05dfe84ec2 [refactor] manually change remaining instances of master key to seed.

jnewbery replied:
done

maflcko reviewed (commented):
nit: The `DeriveNewMasterHDKey` can be done in the scripted-diff I believe.

jnewbery commented:
> The DeriveNewMasterHDKey can be done in the scripted-diff

Done

MarcoFalke commented:
utACK 6249021d152dec348eb4325c0dfccb3ba59f46d1

MarcoFalke commented:
@achow101 and @jonasschnelli Mind to look at the fix, since both of you opened an issue report about this?

jonasschnelli commented:
(re) utACK 79053a5f2b26ee3dfd9a0bb3fd01ac4733fc92b5

MarcoFalke commented on src/wallet/wallet.h:
```diff
@@ -1139,17 +1139,17 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
+    /* Generates a new HD seed (will not be activated) */
+    CPubKey GenerateNewSeed();
 
     /* Derives a new HD master key (will not be activated) */
```
Post merge nit: Should say seed here as well instead of "master key"?

jnewbery replied:
Fixed in #13297
